crypto-sodium/lib/Crypto/Encrypt/Public.hs

```diff
@@ -111,3 +119,16 @@ decrypt
   -> ctBytes -- ^ Encrypted message (cyphertext)
   -> Maybe ptBytes
 decrypt = NaCl.Box.open
+
+
+-- | Generate a new 'SecretKey' together with its 'PublicKey' from a given seed.
+keypairFromSeed
+  :: ByteArrayAccess seed
```

Comment:
According to the docs, seed is not arbitrary, it must be `crypto_box_SEEDBYTES` bytes longs.